### PR TITLE
enhance(deployment): add payload metadata from github

### PR DESCRIPTION
**Pull request by JordanSussman**
Adds the `DeployPayload` field to the `Build` struct to track the payload passed along from GitHub.

xref: https://github.com/go-vela/community/issues/89

### CODE REVIEW

**Review by davidvader — APPROVED** [review]
LGTM

File: raw/map.go [hunk 1]
```diff
@@ -13,6 +14,25 @@ import (
 // StringSliceMap represents an array of strings or a map of strings.
 type StringSliceMap map[string]string
 
+// Value returns the map in JSON format
```
Comment (jbrockopp) [inline raw/map.go:17]: Hate to be nit-picky but could we get `.` to follow this comment?

```suggestion
// Value returns the map in JSON format.
```

File: raw/map.go [hunk 2]
```diff
@@ -13,6 +14,25 @@ import (
...
+func (s StringSliceMap) Value() (driver.Value, error) {
+	value, err := json.Marshal(s)
+	if err != nil {
+		return nil, err
+	}
+	return string(value), nil
+}
+
+// Scan decodes the JSON string into map[string]string
```
Comment (jbrockopp) [inline raw/map.go:26]: Hate to be nit-picky but could we get `.` to follow this comment?

```suggestion
// Scan decodes the JSON string into map[string]string.
```

**Review by jbrockopp — CHANGES_REQUESTED** [review]

**Review by jbrockopp — APPROVED** [review]
LGTM

**Review by kneal — APPROVED** [review]
LGTM 🐬

**Pull request merged**